### DISCUSSION TITLE
loss_weights order in  model.compile

### DISCUSSION THREAD

**Discussion by aoguedao** [lululxvi/deepxde#736] [Q&A]
Hi everyone! 

First of all, thank you very much for this package! I've been spend the last weeks studying it and it is really amazing.

I have a question that maybe could help more people in the near future. The documentation about [model.compile](https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#module-deepxde.model) doesn't say anything how is the order or how many elements need to have the list received as argument of loss_weights. Intuitively, I would say first should be the boundary conditions in the same order as defined and then the elements associated to the PDE residual loss. Am I right?

For example, the [forward second order ODE](https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/ode.2nd.html) utilizes `loss_weights=[0.01,1,1]` since the function 'ode' just return a single element and it has two boundary conditions.

Same idea, the [inverse Poisson equation](https://deepxde.readthedocs.io/en/latest/demos/pinn_inverse/elliptic.inverse.field.html) has `loss_weights=[1, 100, 1000]`, does this mean the first boundary condition has a weight of 1, the second boundary condition (observer) has a weight of 100 and the pde loss a weight of 1000?

Thanks in advance for your help.

**Comment by forxltk** [comment]
No, the boundary loss is always behind the pde loss.

**Reply by aoguedao** [reply]
Hi @forxltk again!  So, is the demo of a forward second order ODE wrong? It says: _"We set the weight of the first initial condition to 0.01, and the weight of the second initial condition and the PDE residual loss to 1"_ with `loss_weights=[0.01,1,1]`

**Reply by forxltk** [reply]
Yes, I think the loss weight in this demo is wrong.

**Reply by lululxvi** [reply]
Yes, it was wrong. See the updated doc.

**Reply by aoguedao** [reply]
Thank you so much. Everything is clear now!

**Reply by jmtucny** [reply]
Has the doc being updated ? I have the same question, and I still see the following text on the website for the [forward Second order ODE](https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/ode.2nd.html) : 

"Now, we have the PDE problem and the network. We build a Model, choose the optimizer, set the learning rate to 0.001, and train the network for 15000 iterations. We set the weight of the ODE loss to 0.01, and the weights of the two ICs to 1. We also compute the L2
 relative error as a metric during training."

With the following code:

`model = dde.Model(data, net)`
`model.compile("adam", lr=.001, loss_weights=[0.01, 1, 1], metrics=["l2 relative error"])`
`losshistory, train_state = model.train(iterations=15000)`

Thanks in advance,

**Reply by lululxvi** [reply]
It is updated.

**Reply by jmtucny** [reply]
Yes, I was confused while reading the documentation. Thank you !
